pkg/controller/blockdeviceclaim/blockdeviceclaim_controller.go

```diff
@@ -270,21 +282,32 @@ func (r *ReconcileBlockDeviceClaim) releaseClaimedBlockDevice(
 	}
 
 	// Check if same deviceclaim holding the ObjRef
+	var claimedBd *apis.BlockDevice
 	for _, item := range bdList.Items {
 		if !r.isDeviceRequestedByThisDeviceClaim(instance, item) {
 			continue
 		}
-
+		claimedBd = &item
 		// Found a blockdevice ObjRef with BlockDeviceClaim, Clear
 		// ObjRef and mark blockdevice released in etcd
-		dvr := item.DeepCopy()
-		dvr.Spec.ClaimRef = nil
-		dvr.Status.ClaimState = apis.BlockDeviceReleased
-		err := r.client.Update(context.TODO(), dvr)
-		if err != nil {
-			klog.Errorf("Error updating ClaimRef of %s: %v", dvr.Name, err)
-			return err
-		}
+		//dvr := item.DeepCopy()
+		//dvr.Spec.ClaimRef = nil
+		//dvr.Status.ClaimState = apis.BlockDeviceReleased
+		//err := r.client.Update(context.TODO(), dvr)
+		//if err != nil {
+		//	reqLogger.Error(err, "Error while updating ObjRef", "BlockDevice-CR:", dvr.ObjectMeta.Name)
+		//	return dvr,err
+		//}
+	}
+	dvr := claimedBd.DeepCopy()
+	dvr.Spec.ClaimRef = nil
+	dvr.Status.ClaimState = apis.BlockDeviceReleased
+
+	r.recorder.Eventf(dvr, corev1.EventTypeNormal, "BlockDeviceCleanUpInProgress", "Released")
```

Comment:
This can be moved to after the `Update()` call, so we can guarantee it has been changed.